Simplenote/Classes/InterlinkProcessor.swift

```diff
@@ -0,0 +1,179 @@
+import Foundation
+
+
+
+// MARK: - InterlinkPresentationContextProvider
+//
+protocol InterlinkProcessorPresentationContextProvider: NSObjectProtocol {
+
+    /// View in the parent's hierarchy that should always appear above the Autocomplete UI
+    ///
+    func parentOverlayViewForInterlinkProcessor(_ processor: InterlinkProcessor) -> UIView
+
+    /// Parent TextView that is triggering the Autocomplete event
+    ///
+    func parentTextViewForInterlinkProcessor(_ processor: InterlinkProcessor) -> UITextView
+
+    /// Parent ViewController!
+    ///
+    func parentViewControllerForInterlinkProcessor(_ processor: InterlinkProcessor) -> UIViewController
+}
+
+
+// MARK: - InterlinkProcessorDatasource
+//
+protocol InterlinkProcessorDatasource: NSObjectProtocol {
+    var interlinkExcudedEntityID: NSManagedObjectID? { get }
+    var interlinkViewContext: NSManagedObjectContext { get }
+}
+
+
+// MARK: - InterlinkProcessor
+//
+class InterlinkProcessor: NSObject {
+
+    private var presentedViewController: InterlinkViewController?
+    private lazy var resultsController = InterlinkResultsController(viewContext: viewContext)
+
+    weak var contextProvider: InterlinkProcessorPresentationContextProvider?
+    weak var datasource: InterlinkProcessorDatasource?
+
+
+    /// Displays the Interlink Lookup UI at the cursor's location when all of the following are **true**:
+    ///
+    ///     1. The Editor isn't Undoing nor Highlighting
+    ///     2. The Editor is the first responder
+    ///     3. There is an interlink `[keyword` at the current location
+    ///     4. There are Notes with `keyword` in their title
+    ///
+    ///  Otherwise we'll simply dismiss the Autocomplete View, if any.
+    ///
+    @objc
+    func processInterlinkLookup() {
+        guard mustProcessInterlinkLookup,
+              let (_, keywordRange, keywordText) = parentTextView.interlinkKeywordAtSelectedLocation,
```

Comment:
Idea: maybe we can already pass bounds for keywords instead of range?

Reply:
Ranges are used for both, keyword positioning (And in a follow up PR) text replacement, I'm afraid.

Even if we do remove the positioning Range, we'd still need the full text range (for replacement purposes), soooo for consistency, I'd vote to keep them both the same. 

WDYT?

Reply:
I see! Let's see how it goes 👍